contractcourt/commit_sweep_resolver.go

```diff
@@ -252,18 +252,86 @@ func (c *commitSweepResolver) Resolve(_ bool) (ContractResolver, error) {
 
 		signDesc = c.commitResolution.SelfOutputSignDesc
 	)
+
+	// commitScriptTimeLockTaproot is a function that returns the timelock
+	// of a non-htlc commitment output.
+	//
+	// TODO(proofofkeags): move this
+	// NOTE FOR REVIEWERS: This function does not really belong here. That
+	// said, I don't know where it should go. If you have opinions, please
+	// comment.
+	commitScriptTimeLockTaproot := func(script []byte) fn.Option[int32] {
+		tok := txscript.MakeScriptTokenizer(0, script)
+		tok.Next() // PUSH <key>
+		if len(tok.Data()) != 32 {
+			return fn.None[int32]()
+		}
+		tok.Next() // CHECKSIG
+		if tok.Opcode() != txscript.OP_CHECKSIG {
+			return fn.None[int32]()
+		}
+
+		tok.Next() // PUSH <timelock>
+		var timelock int32
+		switch {
+		case tok.Opcode() == txscript.OP_0:
+			timelock = 0
+
+		case tok.Opcode() >= txscript.OP_1 &&
+			tok.Opcode() <= txscript.OP_16:
+
+			timelock = int32(tok.Opcode() - txscript.OP_1 + 1)
+
+		case tok.Opcode() == txscript.OP_DATA_1:
+			timelock = int32(tok.Data()[0])
+
+		case tok.Opcode() == txscript.OP_DATA_2:
+			timelock = int32(tok.Data()[0]) |
+				int32(tok.Data()[1])<<8
+
+		case tok.Opcode() == txscript.OP_DATA_3:
+			timelock = int32(tok.Data()[0]) |
+				int32(tok.Data()[1])<<8 |
+				int32(tok.Data()[2])<<16
+
+		case tok.Opcode() == txscript.OP_DATA_4:
+			timelock = int32(tok.Data()[0]) |
+				int32(tok.Data()[1])<<8 |
+				int32(tok.Data()[2])<<16 |
+				int32(tok.Data()[3])<<24
+		default:
+			// We don't cover any data pushes longer than 4 bytes as
+			// that would imply a timelock greater than the history
+			// of Bitcoin.
+			return fn.None[int32]()
+		}
+
+		tok.Next() // CSV
+		if tok.Opcode() != txscript.OP_CHECKSEQUENCEVERIFY {
+			return fn.None[int32]()
+		}
+
+		tok.Next() // DROP
+		if tok.Opcode() != txscript.OP_DROP {
+			return fn.None[int32]()
+		}
+
+		return fn.Some(timelock)
+	}
+
 	switch {
 	// For taproot channels, we'll know if this is the local commit based
-	// on the witness script. For local channels, the witness script has an
-	// OP_DROP value.
-	//
-	// TODO(roasbeef): revisit this after the script changes
-	//  * otherwise need to base off the key in script or the CSV value
-	//  (script num encode)
+	// on the timelock value. For remote commitment transactions, the
+	// witness script has a timelock of 1.
 	case c.chanType.IsTaproot():
-		scriptLen := len(signDesc.WitnessScript)
-		isLocalCommitTx = signDesc.WitnessScript[scriptLen-1] ==
-			txscript.OP_DROP
+		timelock := commitScriptTimeLockTaproot(signDesc.WitnessScript)
+		n, err := timelock.UnwrapOrErr(
```

Comment:
So in this case, we actually _already_ have the relative delay, if we're comfortable _only_ matching on that. In this case, it's the value of `MaturityDelay` (see below). 

That was a stop gap solution I implemented, but I don't think it's sufficient, as a remote party can give us a 1 CSV delay. We have a min+max we'll send, but no min w.r.t what we'll accept. Therefore, the pattern match on the pubkey pushed is required.

Reply:
Happy to implement it with key discrimination, but see my other comment about how we are able to get the relevant contextual information. The key's role isn't a property of the key on its own so we'll need to pull it in some other way.

Reply:
See the main comment above, resolving this.